### PR TITLE
Feature/messaging upgradeable

### PR DESCRIPTION
**Pull request by ChiTimesChi**
## Description
- `MessageBus*` contracts are now upgradeable.
- Added new tests. Setup is to deploy OZ's `TransparentUpgradeableProxy` with `MessageBusUpgradeable` as implementation, and do whatever testing was done for the immutable contracts.

## Checklist

- [ ] New Contracts have been tested
- [ ] Lint has been run
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by ChiTimesChi** [comment]
Should I nuke the old `MessageBus` immutable contracts as well? if so, the deploy script should also be adjusted.

File: test/messaging/MessageBusReceiverUpgradeable.t.sol [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        );
+        messageBusReceiver = MessageBusUpgradeable(address(proxy));
+        messageBusReceiver.initialize(address(0), address(authVerifier));
+    }
+
+    function addressToBytes32(address _addr) public pure returns (bytes32) {
+        return bytes32(uint256(uint160(_addr)));
+    }
+
+    function testAddressToBytes32() public {
```
Comment (BlazeWasHere) [inline test/messaging/MessageBusReceiverUpgradeable.t.sol:33]: test not completed?

File: test/messaging/MessageBusSenderUpgradeable.t.sol [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            receiverAddress,
+            gasFeePricingTest.expectedDstChainId(),
+            bytes(""),
+            bytes("")
+        );
+        messageBusSender.withdrawGasFees(payable(address(1000)));
+        assertEq(address(1000).balance, estimatedFee);
+    }
+
+    function testAddressABIEncode() public {
```
Comment (BlazeWasHere) [inline test/messaging/MessageBusSenderUpgradeable.t.sol:169]: ditto

File: test/messaging/MessageBusReceiverUpgradeable.t.sol (hunk 1 above)
Comment (ChiTimesChi) [reply test/messaging/MessageBusReceiverUpgradeable.t.sol:33]: @BlazeWasHere this is a copypasta from immutable contracts testing. I believe it's just a logging test. In any case, improved them a bit in [732250d](https://github.com/synapsecns/synapse-contracts/pull/149/commits/732250d00946b132e616fffc919657a4aa8a4053):
```js
[PASS] testAddressToBytes32() (gas: 4067)
Logs:
  bytes32: 0x000000000000000000000000de03e73c3785ce086ca85c6315df376a4a64c84b
  address: 0xde03e73c3785ce086ca85c6315df376a4a64c84b
[PASS] testAddressABIEncode() (gas: 9631)
Logs:
    abi.encode: 0x0000000000000000000000006f4e8eba4d337f874ab57478acc2cb5bacdc19c9
       bytes32: 0x0000000000000000000000006f4e8eba4d337f874ab57478acc2cb5bacdc19c9
  encodePacked: 0x6f4e8eba4d337f874ab57478acc2cb5bacdc19c9
       address: 0x6f4e8eba4d337f874ab57478acc2cb5bacdc19c9
```

**Comment by ChiTimesChi** [comment]
Closed in favor of #157

**Pull request closed without merging**